layouts/shortcodes/tabpane.html

```diff
@@ -78,16 +88,23 @@
       {{ $rightpush = . -}}
     {{ end -}}
 
-    {{/* Replace space and +, not valid for css selectors */ -}}
-    {{ $lang := replaceRE "[\\s+]" "-" $lang -}}
+    {{/* Replace by "-" all chars that are not valid in a CSS class name: */ -}}
+    {{ $persistKey = replaceRE "[^a-zA-Z0-9_-]" "-" $persistKey -}}
     <li class="nav-item{{ if $rightpush }} ms-auto{{ end -}}">
       {{/* Generate the IDs for the <a> and the <div> elements */ -}}
       {{ $tabid := printf "tabs-%02v-%v-tab" $Ordinal $index | anchorize -}}
       {{ $entryid := printf "tabs-%02v-%v" $Ordinal $index | anchorize -}}
 
-      <button class="nav-link{{ if and ( not $activeSet ) ( not $disabled ) }} active{{ end }}{{ if $disabled }} disabled{{ end }}{{ if ne $lang "" }}{{ if and $persistLang (not $duplicate) }} persistLang-{{- $lang -}}{{ end }}{{ end }}"
+      <button class="nav-link
+        {{- if and ( not $activeSet ) ( not $disabled ) }} active{{ end -}}
+        {{ if $disabled }} disabled{{ end -}}
+        {{ with $persistKey -}}
+          {{ if $persistTab }} persistLang-{{ . }}{{ end -}}
+        {{ end }}"
           id="{{ $tabid }}" data-bs-toggle="tab" data-bs-target="#{{ $entryid }}" role="tab"
-          {{ if ne $lang "" }}{{ if and $persistLang (not $duplicate) }}onclick="persistLang({{ $lang }});"{{ end }}{{ end -}}
+          {{ with $persistKey -}}
+            {{ if $persistTab }}onclick="persistLang({{ . }});" {{ end -}}
```

Comment:
Since we do now persist the tab rather than the lang, we should consider renaming function `persistLang` to `persistTab`. Of course we then have to change function name in `static/js/tabpane-persist.js`, too.

Reply:
I'd like to address that in a following PR, as noted in https://github.com/google/docsy/pull/1602#issuecomment-1621608047.